content/articles/getting-started-with-ray-tracing-algorithm-in-machine-learning/index.md

```diff
@@ -0,0 +1,72 @@
+
+The process aims to bring real light to the simulation by imitating the behavior of light intensity. It is done by reflecting the light rays from the camera, ‘following’ them until they encounter an object, and counting where each ray of light cuts an object in the square. This then allows it to calculate how the pixel is affected by the light rays, and to keep track of the ray as it jumps on the scene. This requires the computer to track a certain number of rays from each intersection to the reflected light model, thus creating a growing number of rays in the square.
+
+ ![Ray tracing](/engineering-education/getting-started-with-ray-tracing-algorithm-in-machine-learning/tracing.png)
+
+### Types of Ray Tracing Algorithm
+Various methods are induced for effectiveness of Rar Tracing. There are three types of ray tracing algorithms, they include; hybrid ray tracing algorithm, forward ray tracing algorithm, backward ray tracing algorithm.
+
+#### Hybrid Ray Tracing Algorithm
+It's a combination of backward and forward ray tracing techniques hence making it more effective since it has the benefits of both techniques.   
+
+A first pass through the scene is rendered using a basic raster image processing system such as [Vulkan](https://vulkan-tutorial.com/). The first pass is then followed by a ray tracing phase which adds more information to the image created in the first pass.
+
+#### Forward Ray Tracing Algorithm
+It can also be referred to as light ray tracing. This is because it follows photons from the source of light to the specific object. It's the best technique for determining the color of the object despite it being the most inefficient method due to the fact that several rays from the source of light never make it through the view plane and into the eye.
+
+Here, an assumption is made that the reflected ray intersects the eye's surface even though in reality rays are reflected in every direction. Light from the source will travel in a straight direction until it hits a surface where it then changes to other likely directions. 
+
+
+#### Backward Ray Tracing Algorithm 
+In backward ray tracing, an eye ray is produced that travels through the view plane and into space. The eye should be in contact with the object for it to be hit by the eye ray hence being from the point of view plane. The ray tracer then finds out the exact coloring and shade of this point in the view plane after the light beam bounces around. 
+
+Like forward ray tracing, backward ray tracing also has an assumption. It's assumed that only light rays that are traveling through the view plane and then into the eye to produce the final image. The assumption may not be true since many light rays may travel through the view plane in some instances. Backward ray tracing is also known as eye-tracing.
+

```

Comment:
You have not provided a section for 'applications of ray tracing algorithm as per my initial suggestion. add that.